docs/SETUP_TLS_CERTIFICATES.md

```diff
@@ -0,0 +1,232 @@
+
+## Obtaining self-signed certificates
+
+This section describes how to obtain certificates signed by a local CA.
+
+In the course of this guide user will establish a local CA, becoming able to establish local chains of trust. Thus the user will be able to generate certificates for Snap plugins and Snap daemon (`snapteld`) that all share common trust root.
+
+### Remarks
+
+- CA private key needs to be protected, possibly on a dedicated machine or media.
+- Obtained certificates won't be valid outside local setup.
+
+### Procedure
+
+1. Install tool [certstrap](https://github.com/square/certstrap) for generating test certificates. Further steps will assume that `certstrap` is available under `$PATH` location.
+1. Generate root CA certificate:
+    ```
+    certstrap init --cn "snaptest-ca" --o "snaptest" --ou "ca" --key-bits 2048 --years 1 --passphrase '<ca-secure-passphrase>'
+    ```
+    Relevant files:
+    * `./out/snaptest-ca.key` - CA's private key, encrypted with `<ca-secure-passphrase>`; needs to be protected from unauthorized access,
+    * `./out/snaptest-ca.crt` - CA's public certificate, (a certificate root).
+1. Generate server certificate and key to use by plugins:
+    ```
+    certstrap request-cert --cn "snaptest-srv" --ip "127.0.0.1" --domain "localhost" --passphrase '' --key-bits 2048 --o "snaptest" --ou "server"
+    certstrap sign "snaptest-srv" --CA "snaptest-ca" --passphrase '<ca-secure-passphrase>' --years 1
+    ```
+    This will produce a private key certificate file necessary to establish secure TLS channel on server. Server's private key file should not be passphrase-protected (or each plugin would have to ask for passphrase on start).
+    Relevant files:
+    * `./out/snaptest-srv.key` - server's private key,
+    * `./out/snaptest-srv.crt` - server's public certificate.
+1. Generate client certificate and key to use by framework:
+    ```
+    certstrap request-cert --cn "snaptest-cli" --ip "127.0.0.1" --domain "localhost" --passphrase '' --key-bits 2048 --o "snaptest" --ou "client"
+    certstrap sign "snaptest-cli" --CA "snaptest-ca" --passphrase '<ca-secure-passphrase>' --years 1
+    ```
+    This will produce a private key certificate file necessary to securely connect to plugins. Client's private key file should not be passphrase-protected.
+    Relevant files:
+    * `./out/snaptest-cli.key` - client's private key,
+    * `./out/snaptest-cli.crt` - client's public certificate.
+    
+Files that need to be given at plugin startup:
+* `snaptest-ca.crt` - CA's certificate,
+* `snaptest-srv.key` - server's private key file,
+* `snaptest-srv.crt` - server's certificate file.
+
+Files that need to be given at `snapteld` startup:
+* `snaptest-ca.crt` - CA's certificate,
+* `snaptest-cli.key` - framework's private key file,
+* `snaptest-cli.crt` - framework's certificate file.
+
+## Obtaining legitimate certificate from a public CA Let's Encrypt
```

Comment:
Grammatically, either need a colon (":") between CA and Let's Encrypt or remove it from the title if it's not important where it's from for the example.